pkg/template/api/types.go

```diff
@@ -7,15 +7,22 @@ import (
 
 // Template contains the inputs needed to produce a Config.
 type Template struct {
-	kapi.TypeMeta   `json:",inline"`
-	kapi.ObjectMeta `json:"metadata,omitempty"`
-
-	// Required: A list of resources that might reference parameters
-	Items []runtime.Object `json:"items"`
+	kapi.TypeMeta
+	kapi.ObjectMeta
 
 	// Optional: Parameters is an array of Parameters used during the
 	// Template to Config transformation.
-	Parameters []Parameter `json:"parameters,omitempty"`
+	Parameters []Parameter
+
+	// Required: A list of resources to create
+	Objects []runtime.Object
```

Comment:
if you're reworking this anyway, any reason not to do the Spec nesting you proposed earlier?